### PR TITLE
fix: able to delete the only version

### LINKED ISSUE #4978: able to delete the only version in the app
### What is the current behaviour?

Currently, when there is only single version on the app, we are able to delete it.

### What is the expected behaviour?

Ideally, the user should not be able to delete the last/only version in the app

### How to reproduce the issue?

1. Create an app with a version
2. delete the version from the version manager

### Screenshots or Screencast

https://user-images.githubusercontent.com/25361949/207345333-5ef2bb3f-9720-4e4b-827e-c7c1b0ea77bc.mov

### PR DESCRIPTION
**Pull request by mikebarr24**
fixes #4978

### CODE REVIEW

File: server/src/controllers/apps.controller.ts [hunk 1]
```diff
@@ -284,8 +284,12 @@ export class AppsController {
   @UseGuards(JwtAuthGuard)
   @Delete(':id/versions/:versionId')
   async deleteVersion(@User() user, @Param('id') id, @Param('versionId') versionId) {
+    const number = await this.appsService.fetchVersions(user, id);
     const version = await this.appsService.findVersion(versionId);
     const ability = await this.appsAbilityFactory.appsActions(user, id);
+    if (number.length <= 1) {
```
Comment (gsmithun4) [inline server/src/controllers/apps.controller.ts:290]: Shall we add this after the ability checks?
https://github.com/ToolJet/ToolJet/pull/5123/files#diff-4cc88c767a6288a89c3d84175bac75d447c6fda36b23275ffcd88f323d2e979bR294

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: server/src/controllers/apps.controller.ts (hunk 1 above)
Comment (mikebarr24) [reply server/src/controllers/apps.controller.ts]: I think that should be it now.

**Review by gsmithun4 — DISMISSED** [review]

**Comment by gsmithun4** [comment]
@mikebarr24 Please resolve conflicts

**Comment by mikebarr24** [comment]
Sorry. Don't think this is right and don't have time just now to fix.

**Comment by shubh22** [comment]
Task linked: [CU-85zrht8vb Able to delete all the versions. and the app is crashing when exporting the app without any version.](https://app.clickup.com/t/85zrht8vb)

**Pull request merged**